### ISSUE TITLE
Terminal Font Detection: Add support for more terminal emulators.

### ISSUE THREAD

**Issue by dylanaraps** [dylanaraps/neofetch#270]
Terminal font detection was just pushed to master with support for a handful of
terminal emulators. This issue aims to add support for more terminal emulators.

**Profile Based Configs:**

We need to figure out how to get the font on a per profile basis.
- [ ] Gnome-terminal
- [ ] iTerm2
- [x] Konsole
- [ ] Terminator
- [x] Terminology

**No Configs:**

Font support can't be added as these terminals are configured at compile time.
- [ ] st

If your terminal isn't listed here, comment below. :)

**Comment by kelaun** [comment]
For terminator:
The font can be read in the ~/.config/terminator/config file.
The needed line starts with: 
`  title_font = `
An example: 
`  title_font = Iosevka Term 10`

**Comment by dylanaraps** [comment]
The problem is that `terminator` uses a profile based config, here's a snippet
from a terminator config I just generated.

```sh
  [[Profile1]]
    font = Roboto Mono weight=250 10
  [[Profile2]]
    font = lemon
```
Notice how `font =` appears twice, once for each profile the user has created.

Like the other profile based config terminals we need to first figure out which 
profile is currently in use so that we can find the right font.
 
I don't want to just randomly grab a font from the config either since we'll then
be displaying incorrect info.

**Comment by CadaverLab** [comment]
What about `termite`? You can specify font in the config file (~/.config/termite/config) like so

`[options]`
`font = Iosevka Term 10`

**Comment by dylanaraps** [comment]
Termite support is already in master, see this part of the script:

https://github.com/dylanaraps/neofetch/blob/master/neofetch#L1572-L1574

**Comment by zAMLz** [comment]
Is it possible to avoid reading fonts that have been commented out in the termite config?
For example

>;font = envypn, regular, 13
>;font = tamsyn, regular,13
>;font = tamzenforpowerline, 11
>font = tamzenforpowerline, 13

Neofetch says envypn is my current font, when its actually tamzenforpowerline.

**Comment by dylanaraps** [comment]
Yes!

I'll be home in an hour and I'll add this then. Thanks for finding this
bug. :+1:

**Comment by zAMLz** [comment]
No problem, glad I could help!

**Comment by dylanaraps** [comment]
Fixed.

**Comment by zAMLz** [comment]
Nice mate!

**Comment by tabakisp** [comment]
Hello, 
`xfce4-terminal` is detecting the wrong font. According to `{XDG_CONFIG_HOME}/xfce4/terminal/terminalrc`

> FontName=Menlo 9

But Neofetch shows 

> Noto Sans 9 [GTK3]

**Comment by dylanaraps** [comment]
There are two font outputs, `terminal font` and `gtk/qt font`. The Noto sans output is from GTK/QT and not your terminal.

**Comment by tabakisp** [comment]
You are absolutely right but I don't use `Noto Sans` on my system. 
My font in Appearance and Window Manager is `SFNS Display`.

**Comment by dylanaraps** [comment]
Can I see a screenshot?

**Comment by tabakisp** [comment]
![screenshot_2016-10-13_15-30-16](https://cloud.githubusercontent.com/assets/7551798/19349222/9a0711d8-915a-11e6-94dd-37f05a76d39f.png)![screenshot_2016-10-13_15-30-36](https://cloud.githubusercontent.com/assets/7551798/19349223/9a08172c-915a-11e6-813b-ee5b7bb480e0.png)
![screenshot_2016-10-13_15-35-37](https://cloud.githubusercontent.com/assets/7551798/19349269/d06e53a8-915a-11e6-9739-77c1e1bfc650.png)
![screenshot_2016-10-13_15-31-15](https://cloud.githubusercontent.com/assets/7551798/19349224/9a09d242-915a-11e6-8759-8e6e207477c6.png)

**Comment by dylanaraps** [comment]
Try out the latest master.

**Comment by tabakisp** [comment]
Boom goes the dynamite. Tyvm.

**Comment by mstraube** [comment]
Hi, I've done the [font detection for Konsole](https://github.com/mstraube/neofetch/commit/9a4054e19db8838ad5827d5fd5cff83b3c0fb1ec).

It uses qdbus to get the active profile of the window/tab from which neofetch is executed and then greps the font name from the profile config file.

**Comment by dylanaraps** [comment]
Awesome, I've started writing some comments.

**Comment by mstraube** [comment]
This 84579a298e5bb1326cb73de36097cb2f852c7d22 adds support for Guake. 

The get_term() function didn't recognise guake at all, because get_process_name() returned python2 on my Arch system. `ps x | grep guake` looks like this: `32682 ?        Sl     0:02 python2 -m guake.main`

Currently it sets term_font only if `Use system default font` is unchecked in the Guake settings.

What do you think?

**Comment by dylanaraps** [comment]
Awesome!

I left some comments. The font part looks good, just the detection of `guake` needs a rethink.

**Comment by d3rrial** [comment]
Maybe someone can add support for alacritty?
https://github.com/jwilm/alacritty

Right now the config files are not xdg compliant, but they should be soon, so it might make sense to do it then (although since it's a very new terminal emulator, the popularity might not be incredibly high)

**Comment by dylanaraps** [comment]
I'm currently building alacritty. :+1:

**Comment by dylanaraps** [comment]
Closed in favor of #613
